python/ray/air/predictor.py

```diff
@@ -32,18 +70,57 @@ def from_checkpoint(cls, checkpoint: Checkpoint, **kwargs) -> "Predictor":
         """
         raise NotImplementedError
 
+    @DeveloperAPI
+    @abc.abstractmethod
+    def _predict_pandas(self, data: pd.DataFrame, **kwargs) -> pd.DataFrame:
+        """Perform inference on a Pandas DataFrame.
+
+        All predictors are expected to implement this method.
+
+        Args:
+            data: A pandas DataFrame to perform predictions on.
+            kwargs: Arguments specific to the predictor implementation.
+
+        Returns:
+            A pandas DataFrame containing the prediction result.
+
+        """
+        raise NotImplementedError
+
+    @PublicAPI(stability="alpha")
     def predict(self, data: DataBatchType, **kwargs) -> DataBatchType:
         """Perform inference on a batch of data.
 
         Args:
-            data: A batch of input data. Either a pandas Dataframe or numpy
-                array.
-            kwargs: Arguments specific to predictor implementations.
+            data: A batch of input data of type ``DataBatchType``
+            kwargs: Arguments specific to predictor implementations. These are passed
+                directly to predict_pandas.
 
         Returns:
             DataBatchType: Prediction result.
         """
-        raise NotImplementedError
+
+        pass
+
+        # TODO(amogkam): Implement below code.
+        # data_df = self._convert_to_pandas(data)
+        #
+        # if hasattr(self, "preprocessor") and self.preprocessor:
+        #     data_df = self.preprocessor.transform_batch(data_df)
+        #
+        # predictions_df = self.predict_pandas(data_df)
+        # return self._convert_from_pandas(predictions_df, type=type(data))
+
+    def _convert_to_pandas(self, data: DataBatchType) -> pd.DataFrame:
+        """Convert the provided data to a Pandas DataFrame."""
+        pass
+
+    def _convert_from_pandas(
```

Comment:
Nit: I'd vote for `_convert_to_batch_type` or `_convert_pandas_to_batch_type` rather than `_convert_from_pandas`.

Reply:
Ok updated!